include/aspect/material_model/reactive_fluid_transport.h

```diff
@@ -0,0 +1,153 @@
+         * @p in and @p melt_fractions need to have the same size.
+         *
+         * @param in Object that contains the current conditions.
+         * @param melt_fractions Vector of doubles that is filled with the
+         * allowable free fluid fraction for each given input conditions.
+         */
+        virtual void melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
+                                     std::vector<double> &melt_fractions) const override;
+
+        /**
+         * Initialize the base model at the beginning of the model run
+         * @copydoc MaterialModel::Interface::initialize()
+         */
+        void
+        initialize () override;
+
+        /**
+         * Update the base model at the beginning of each timestep.
+         */
+        void update() override;
+
+        /**
+         * @copydoc MaterialModel::Interface::evaluate()
+         */
+        void
+        evaluate (const typename Interface<dim>::MaterialModelInputs &in,
+                  typename Interface<dim>::MaterialModelOutputs &out) const override;
+
+        /**
+         * @copydoc MaterialModel::Interface::declare_parameters()
+         */
+        static void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * @copydoc MaterialModel::Interface::parse_parameters()
+         */
+        void
+        parse_parameters (ParameterHandler &prm) override;
+
+        /**
+         * If this material model can produce additional named outputs
+         * that are derived from NamedAdditionalOutputs, create them in here.
+         */
+        virtual
+        void
+        create_additional_named_outputs (MaterialModel::MaterialModelOutputs<dim> &out) const override;
+
+      private:
+
+        /**
+         * Pointer to the material model used as the base model
+         */
+        std::unique_ptr<MaterialModel::Interface<dim>> base_model;
+
+        // Variables that describe the properties of the fluid, i.e. its density,
+        // viscosity, and compressibility.
+        // Properties of the solid are defined in the base model.
+        double reference_rho_f;
+        double eta_f;
+        double fluid_compressibility;
+
+        // Material properties governing the transport of the fluid with respect
+        // to the solid, i.e., the bulk viscosity (relative to the shear viscosity),
+        // the permeability, and how much the solid viscosity changes in the presence
+        // of fluids.
+        double shear_to_bulk_viscosity_ratio;
+        double reference_permeability;
+        double alpha_phi;
+
+        // Time scale for fluid release and absorption.
+        double fluid_reaction_time_scale;
+
+        /**
+         * Enumeration for selecting which type of scheme to use for reactions between.
+         * fluids and solids. The only current option is a scheme where no reactions occur.
```

Comment:
That is not true. Your scheme has a zero solubility, but there will be reactions if there is bound water at the start. So you either need to change the implementation (if you actually want no reactions) or the name of the input parameter and the documentation (to something like `zero_solubility`). 

Or you add both, and the you actually have a choice between the two!